src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { forwardRef } from 'react'
+import { CSS, StitchesVariants, styled } from 'stitches.config'
+import {
+  Root,
+  Trigger,
+  Content,
+  Item,
+  CheckboxItem,
+  ItemIndicator,
+} from '@radix-ui/react-dropdown-menu'
+import { Slot } from '@radix-ui/react-slot'
+import { Check } from 'index'
+
+const itemStyles = {
+  fontSize: '$0',
+  padding: '$2 $3',
+  cursor: 'default',
+  '&::before': {
+    boxSizing: 'border-box',
+  },
+  '&::after': {
+    boxSizing: 'border-box',
+  },
+
+  borderRadius: '$default',
+  backgroundColor: 'transparent',
+  border: 'none',
+  transition: 'background 0.5s',
+  '&:focus': {
+    background: '$grey4',
+    cursor: 'pointer',
+  },
+}
+
+const StyledContent = styled(Content, {
+  minWidth: 130,
+  backgroundColor: '$paper',
+  borderRadius: '$default',
+  padding: '$1',
+  boxShadow: '$2',
+})
+
+const StyledItemIndicator = styled(ItemIndicator, {
+  position: 'absolute',
+  left: 5,
+})
+
+const DEFAULT_TAG = 'div'
+
+/**
+ * StyledMenu base component
+ */
+const StyledMenu = styled(Root, {})
+const StyledCheckboxItem = styled(CheckboxItem, itemStyles)
+
+const StyledItem = styled(Item, itemStyles)
+
+type MenuCSSProp = { css?: CSS }
+type MenuVariants = StitchesVariants<typeof StyledMenu>
+type MenuOwnProps = MenuCSSProp & MenuVariants
+
+type MenuComponent = Polymorphic.ForwardRefComponent<
+  typeof DEFAULT_TAG,
+  MenuOwnProps
+>
+
+/**
+ * Menu component
+ */
+export const Menu = forwardRef(({ ...props }, ref) => (
+  <StyledMenu ref={ref} {...props} />
+)) as MenuComponent
+Menu.toString = () => `.${StyledMenu.className}`
+
+type MenuContentCSSProp = { css?: CSS }
+type MenuContentOwnProps = Polymorphic.OwnProps<typeof Content> &
+  MenuContentCSSProp & {}
+
+type MenuContentComponent = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Content>,
+  MenuContentOwnProps
+>
+
+export const MenuContent = forwardRef(
+  ({ children, css, ...props }, forwardedRef) => (
+    <StyledContent css={css as CSS} {...props} ref={forwardedRef}>
+      {children}
+    </StyledContent>
+  )
+) as MenuContentComponent
+
+type MenuTriggerCSSProp = { css?: CSS }
+type MenuTriggerOwnProps = Polymorphic.OwnProps<typeof Trigger> &
+  MenuTriggerCSSProp
+type MenuTriggerComponent = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Trigger>,
+  MenuTriggerOwnProps
+>
+
+export const MenuTrigger = forwardRef(
+  ({ children, ...props }, forwardedRef) => (
+    <Trigger as={Slot} {...props} ref={forwardedRef}>
+      {children}
+    </Trigger>
+  )
+) as MenuTriggerComponent
+
+type MenuItemCSSProp = { css?: CSS }
+type MenuItemOwnProps = Polymorphic.OwnProps<typeof StyledItem> &
+  MenuItemCSSProp
+type MenuItemComponent = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof StyledItem>,
+  MenuItemOwnProps
+>
+
+export const MenuItem = forwardRef(({ children, ...props }, forwardedRef) => (
```

Comment:
Genuinely do know if this forwardRef ids required here. If we are not actually doing anything but forwarding the props we might be ok just to 

```
export const MenuItem = StyledItem
```

Or just call StyledItem, MenuItem above and export.

Two deciding factors would be does it work with out it (I would hope so as I expect the styled to do the appropriate ref forwarding). Second even if it does work with out it do we get the correct props in the storybook. 

Anyway no real harm in doing it if it's not required, just lib size.

Reply:
VSCode and Storybook is happy but building seems to give an error `Menu.tsx(108,14): semantic error TS2742: The inferred type of 'MenuItem' cannot be named without a reference to '@radix-ui/react-avatar/node_modules/@radix-ui/react-primitive'. This is likely not portable. A type annotation is necessary.`

Reply:
Turns out this was an issue with multiple versions of @radix-ui/primitive. Updating @radix-ui/react-accordion fixed the issue.